code/datums/components/fertile_egg.dm

```diff
@@ -0,0 +1,67 @@
+/**
+ * # A fertile egg component!
+ *
+ * This component tracks over time if the atom is in ideal conditions,
+ * and eventually hatches into the embryonic type.
+ *
+ * The initial design of this component was to make more generic the code for
+ * chickens laying eggs.
+ */
+
+
+/datum/component/fertile_egg
+	/// What will come out of the egg when it's done.
+	var/embryo_type
+
+	/// Minimum growth rate per tick
+	var/minimum_growth_rate
+
+	/// Maximum growth rate per tick
+	var/maximum_growth_rate
+
+	/// Total growth required before hatching.
+	var/total_growth_required
+
+	/// The current amount of growth.
+	var/current_growth = 0
```

Comment:
should this really be an initial arg?

Reply:
As in, this should just be passed in by the adder of the component, or it should always start at 0 anyway?